Logger/BaseLogger.cs

```diff
@@ -1,7 +1,8 @@
-﻿namespace Logger;
-
-public abstract class BaseLogger
+﻿namespace Logger
 {
-    public abstract void Log(LogLevel logLevel, string message);
+    public abstract class BaseLogger
+    {
+        public abstract string ClassName { get; set; }
```

Comment:
```suggestion
        public string? ClassName { get; set; }
```
This doesn't need to be abstract here, but it's not wrong depending on how you intend this to be used. Also, it's okay to mark this as nullable, you just need to make sure that you're expecting that it can be null.